### ISSUE TITLE
Add support for filter operators

### ISSUE THREAD

**Issue by AbegaM** [thevahidal/soul#105]
# Purpose of the Issue
I would like to add support for filtering operators in Soul. Currently, Soul filters table fields with the `_filters` field, which works fine, but I am unable to filter items using the operators `<`, `>`, `<=`, `>=`, and `!=`. For example, I can fetch an invoice from Soul using the following URL, which works fine:

   ```js
      localhost:8000/api/tables/invoices/rows?_filters=Total:1.98
   ```

However, I might want to fetch invoices where the total value is less than 2 or greater than 2, and I cannot achieve this with Soul's current functionality.

# Proposed Solution
There could be several options to add this feature, but I propose passing double underscores after the column name. Here are some examples:

**Greater than (`>`):** 
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__gt:3
```

**Less than (`<`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__lt:3
```

**Greater than or equal to (`>=`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__gte:3
```

**Less than or equal to (`<=`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__lte:3
```

**Not equal to (`!=`):**
```js
localhost:8000/api/tables/invoices/rows?_filters=Total__neq:3
```

This proposal would allow users to filter table rows using the operators listed above, making Soul's functionality more flexible and powerful.

**Comment by AbegaM** [comment]
Hello @thevahidal 

Could you please review my feature request issue and let me know what you think? I'm planning to start working on a pull request soon. Thank you!"

**Comment by IanMayo** [comment]
Note: this strategy for providing filter operators appears to match the one used in `react-admin`:
https://marmelab.com/react-admin/FilteringTutorial.html#filter-operators

**Comment by thevahidal** [comment]
Thanks for the feature request @AbegaM. This is for sure a must have feature for Soul and I think the way your proposal works is really great!
Please start your PR and let me know if there's anything I can help with it.

**Comment by thevahidal** [comment]
Add support in #107

**Comment by IanMayo** [comment]
BTW - @thevahidal - you can use the repo settings to delete branches once the PR is merged.  It's a  house-keeping chore that helps out just a little.

**Comment by thevahidal** [comment]
Hey @IanMayo, I'll do it for sure.

**Comment by IanMayo** [comment]
Hello @thevahidal - any chance of a new release?  I have a PR in another project that is blocked,  waiting on this feature.

**Comment by IanMayo** [comment]
It appears that the existing filters to not match for `null`.   When we provide `_filters=batch:null`, I suspect `Soul` is doing a string comparison.

So, I believe we need a filter attribute for the SQLite value of null.  Maybe a new `__null` operator?

**Comment by IanMayo** [comment]
Good plan, let's defer release until we can filter for null.

**Comment by thevahidal** [comment]
It seems that @AbegaM started a new PR for the matter https://github.com/thevahidal/soul/pull/117/files, let's release these two together.

**Comment by IanMayo** [comment]
Closed in #117
